### ISSUE TITLE
failure when using this plugin with shadow plugin

### ISSUE THREAD

**Issue by pbthorste** [sebersole/gradle-maven-publish-auth#24]
Hi,

I encountered this error when using the plugin in combination with the shadow plugin:

```
Authentication scheme 'all'(Authentication) is not supported by protocol 'file'
```

Not sure if the problem lies in this plugin or in the shadow plugin. Sample build.gradle:

```
plugins {
    id 'java'
    id "com.github.johnrengelman.shadow" version "5.2.0"
    id 'maven-publish'
    id "org.hibernate.build.maven-repo-auth" version "3.0.0"
}

group 'org.example'
version '1.0-SNAPSHOT'

sourceCompatibility = 1.8

repositories {
    mavenCentral()
    mavenLocal()

}

dependencies {
    compile group: 'org.apache.commons', name: 'commons-lang3', version: '3.9'
    testCompile group: 'junit', name: 'junit', version: '4.12'
}

shadowJar {
    mergeServiceFiles()
}
```

**Comment by sebersole** [comment]
Unfortunately I don't really have time to investigate this issue.  If you can find the cause and it is really with this plugin I am happy to merge a PR with a fix or fix it if the problem is simple

**Comment by coyotesqrl** [comment]
The issue is that `mavenLocal()` is listed as one of the repositories. This plugin should probably not attempt to apply matching credentials to maven repositories with unsupported protocol types (in this case, `file:`.

**Comment by sebersole** [comment]
But the plugin applies credentials based on security info it finds in Maven's settings.xml.  So you are saying that in settings.xml you define credentials for maven-local but you don't want this plugin to apply them?  

Otherwise, I'm not sure how this plugin would even attempt to apply any credentials to maven-local

**Comment by coyotesqrl** [comment]
Okay. I spent a bit more time debugging this morning and have figured out why my PR resolves the problem that _shouldn't exist_ with `mavenLocal`. It isn't the extra effort I've put in to block protocols that can't be authenticated with username:password; it's a side-effect of the way I did it. 

My PR includes a little refactoring, changing the `Helper.applyCredentials()` signature to take a `MavenArtifactRepository` instead of the `repoId` and `PasswordCredentials`. Which means that the callers no longer invoke `getCredentials()` on the repository object. That's been moved in my version to after the call to `locateAuthenticationCredentials()`, once I know that we should bother to apply the credentials from `settings.xml`.

My PR works because `getCredentials()` has the side-effect of creating and setting the credentials object on the repository to a new `PasswordCredentials` instance (https://github.com/gradle/gradle/blob/master/subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/AuthenticationSupporter.java#L51-L59). I don't invoke that except when we should be adding them, so the credentials remain null.

Personally, I don't think it's bad to restrict the plugin as it is now to only supporting those protocols that Gradle allows be protected by username:password, but if you disagree with that addition then the change can be simplified to the refactor so it doesn't call `getCredentials()` until it knows it should.

**Comment by sebersole** [comment]
Its not that I disagree we should restrict it.  I'm just not a huge fan of restricting it based on a finite set of protocol strings.

**Comment by coyotesqrl** [comment]
Unfortunately, I don't see a code-based way of determining whether a protocol supports username:password. That's why I fell back to the documentation and the hard-coded strings. I can remove that restriction from the PR and just leave the refactor. Neither way is going to work as it stands for `s3` or `gcs` and no one should be adding username:password credentials for any file-based repositories anyway.

Your call.

**Comment by sebersole** [comment]
Resolved via #27
